### PR TITLE
Add pagination in react table

### PR DESCRIPTION
**Pull request by bipulthapa**
- Add pagination from server site with correct limit and page number
- Add pagination in the employee list page
- Add data to JSON mock server db.json

### CODE REVIEW

File: src/components/common/pagination/Pagination.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
+import React, { Component } from 'react';
+import _ from 'lodash';
```
Comment (sanjevShakya) [inline src/components/common/pagination/Pagination.js]: I see that you've used lodash range function. Can you research on why we should not import entire lodash?
And then change this to 
```javascript
import _range from 'lodash/range';
```

File: src/components/common/pagination/Pagination.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+const PageButtonComponent = props => <button {...props}>{props.children}</button>;
+
+class Pagination extends Component {
+  state = { visiblePages: [] };
```
Comment (sanjevShakya) [inline src/components/common/pagination/Pagination.js]: we have a convention to add the state inside the constructor function

File: src/components/common/pagination/Pagination.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+  componentWillReceiveProps(nextProps) {
```
Comment (sanjevShakya) [inline src/components/common/pagination/Pagination.js]: the method is deprecated and we should use ```getDerivedStateFromProps```

**Review by sanjevShakya — CHANGES_REQUESTED** [review]

**Comment by sanjevShakya** [comment]
We have redux setup in the project. Shouldn't we used flux architecture for our data management?
@pratishshr can you add something to this.

File: src/components/common/pagination/Pagination.js (hunk 1 above)
Comment (bipulthapa) [reply src/components/common/pagination/Pagination.js]: For the sake of optimization, we should not import the entire lodash. Using specific imports will reduce the bundle size. Thank you dai for pinpointing it

File: src/components/common/pagination/Pagination.js (hunk 3 above)
Comment (bipulthapa) [reply src/components/common/pagination/Pagination.js]: In the latest change, I have use componentDidUpdate. Hope this is also a good practice

File: src/components/common/pagination/Pagination.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+class Pagination extends Component {
+  constructor() {
+    super();
+    this.state = { visiblePages: [] };
+  }
+
+  componentDidUpdate(prevProps) {
+    const { pages, page, visiblePageCount } = this.props;
+
+    if (prevProps.pages != pages) {
```
Comment (bikkimahato) [inline src/components/common/pagination/Pagination.js:15]: Please use strict inequality operator !==

**Pull request closed without merging**